src/strands/session/session_repository.py

```diff
@@ -0,0 +1,51 @@
+"""Session repository interface for agent session management."""
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from ..types.session import Session, SessionAgent, SessionMessage
+
+
+class SessionRepository(ABC):
+    """Abstract repository for creating, reading, and updating Sessions, AgentSessions, and AgentMessages."""
+
+    @abstractmethod
+    def create_session(self, session: Session) -> Session:
+        """Create a new Session."""
+
+    @abstractmethod
+    def read_session(self, session_id: str) -> Optional[Session]:
+        """Read a Session."""
+
+    @abstractmethod
+    def create_agent(self, session_id: str, session_agent: SessionAgent) -> None:
```

Comment:
I think all of these methods (create/update/read agent/message) are part of Session's responsibility. In terms of interfaces, I think these should be `session.create_agent(agent)` for example. Current setup feels a bit counterintuitive to me

I think we should refactor the interfaces a bit, such that:
- current `Session` becomes `SessionData` to store/serialize/deserialize session data
- `Session` abstract class is created for updating session.
- `SessionRepository` is removed and responsibilities are split
  - session related methods go to `Session`
  - create and read session methods go to `SessionManager`

I think that makes the interface a bit simpler and more intuitive
- Session Repository is gone. 2 responsibilities are split